src/app/sources/[sourceId]/UpdateSourceExtractedRecipeForm.tsx

```diff
@@ -6,14 +6,19 @@ import fetcher from "@/lib/fetcher";
 import Textarea from "@/components/elements/Textarea";
 import useSWRImmutable from "swr/immutable";
 import { treeifyError } from "zod";
+import { useActionState } from "react";
+import { updateSourceExtractedRecipeAction } from "@/lib/actions/sources";
 
-export default function FetchExtractedRecipeForm({
+export default function UpdateSourceExtractedRecipeForm({
   sourceId,
-  formAction,
 }: {
   sourceId: string;
-  formAction: (formData: FormData) => Promise<void> | void;
 }) {
+  const [state, formAction, pending] = useActionState(
+    updateSourceExtractedRecipeAction,
+    undefined,
+  );
```

Comment:
We're using `useActionState` all over the place to help with form submissions. I hate this `undefined` but it works and is type safe... somehow...